helix-loader/src/grammar.rs

```diff
@@ -146,14 +238,21 @@ where
 
     drop(tx);
 
-    // TODO: print all failures instead of the first one found.
-    rx.iter()
-        .find(|result| result.is_err())
-        .map(|err| err.with_context(|| format!("Failed to {} some grammar(s)", action)))
-        .unwrap_or(Ok(()))
+    rx.iter().collect()
 }
 
-fn fetch_grammar(grammar: GrammarConfiguration) -> Result<()> {
+struct FetchSuccess {
+    grammar_id: String,
+    action: FetchAction,
```

Comment:
This is more of a `status` than an `action` since it describes what happened in the fetch

Reply:
Yes, thanks for the name